### PR TITLE
Implement logic for tournament treasury account id derivation

### PR DESCRIPTION
**Pull request by clangenb**

### CODE REVIEW

File: pallets/ajuna-tournament/src/lib.rs [hunk 1]
```diff
@@ -182,14 +182,15 @@ pub mod pallet {
 
 		/// The account ID of the sub-account for a given season_id/tournament_id.
 		pub fn tournament_treasury_account_id(
-			season_id: &T::SeasonId,
-			tournament_id: &TournamentId,
+			season_id: T::SeasonId,
+			tournament_id: TournamentId,
```
Comment (clangenb) [inline pallets/ajuna-tournament/src/lib.rs:186]: If we dereference the values inside, I prefer not having borrowed values as function args.

**Comment by DidacSF** [comment]
I like the idea! Now the problem though is that for testing this doesn't work, that's because the AccountId is only 8 bytes in the testing runtime, so the bytes get truncated at the PalletId, meaning all accounts are the same.

How would you solve this? Maybe `#cfg[test]` flag for a mock implementation in tests? Or is there a better solution?

**Comment by clangenb** [comment]
Hmm, is there a specific reason that the account ID is only 8 bytes? If we want to reflect the actual production runtime behavior, I think we should make the account ID in this case be the same as the in the production runtime. TBH, I think this would be the best solution, although I haven't researched the consequences of such a change.

Otherwise, I think we could make the `TreasuryAccount` an associated type and create an associated type for its constructor method. I prefer if `cfg(test)` is only used to add functionality; changing functionality is dangerous in my opinion, as you no longer test the production behavior.

**Comment by DidacSF** [comment]
Added the change so that `MockAccountId` is more aligned with the expected production type, with this all tests pass, so I'll merge if you agree @clangenb

**Comment by clangenb** [comment]
Yes, sounds perfect!

**Pull request merged**